### PR TITLE
Send email notifications when tool jobs finish

### PR DESCRIPTION
**Pull request by vlagrassa**
Currently, only sends to admin users, when Nemascan or Heritability jobs finish.

Introduces new secret values:
- `NO_REPLY_EMAIL` - email to send notifications from 
- `CAENDR_API_SITE_ACCESS_TOKEN` - shared secret between site and pipeline API

Uses env vars:
- `MODULE_SITE_BASE_URL` - to determine where to send the API call. Was previously only defined for `main`.

Adds compound launch config for site + pipeline API, to test flow of information back and forth between the two.

### CODE REVIEW

File: src/pkg/caendr/caendr/services/cloud/lifesciences.py [hunk 1]
```diff
@@ -127,5 +137,58 @@ def update_all_linked_status_records(kind, operation_name):
       logger.warn(f"Unrecognized kind: {kind}")
       continue
 
+    # Only send a notification if the report's status has not been updated yet
+    # TODO: Should be able to remove kind check if all report notifications merged into one system
+    should_send_notification = done \
```
Comment (r-vieira) [inline src/pkg/caendr/caendr/services/cloud/lifesciences.py]: consider refactoring into array of conditions. 
```python
done = True
other = (1 == 0)

conditions = [ 
  done, 
  other 
]
should_send = len([ c for c in conditions if c ]) == len(conditions)
```

File: src/pkg/caendr/caendr/services/cloud/lifesciences.py [hunk 2]
```diff
@@ -127,5 +137,58 @@ def update_all_linked_status_records(kind, operation_name):
...
+    logger.debug(f'Should send notification for report {status_record.id}: {should_send_notification}. (done = {done}, kind = {kind}, current status = {status_record["status"]})')
+
+    # Update the report status
     status_record.set_properties(status=status)
     status_record.save()
+
+    # Send the notification, if applicable
+    # In theory, doing this after the database update should prevent duplicate emails
+    # For now, only send email notifications to admin users
+    if should_send_notification:
```
Comment (r-vieira) [inline src/pkg/caendr/caendr/services/cloud/lifesciences.py]: consider bailing out early if `should_send_notification` is False. 
```python
if not should_send_notification:
  return 
```

File: src/pkg/caendr/caendr/services/cloud/lifesciences.py [hunk 3]
```diff
@@ -127,5 +137,58 @@ def update_all_linked_status_records(kind, operation_name):
...
+    },
+  )
+
+  # Get the JSON body from the request
+  if response.status_code != 200:
+    return response
+  message = response.json()
+
+  # Send the email
+  return send_email({
```
Comment (r-vieira) [inline src/pkg/caendr/caendr/services/cloud/lifesciences.py:195]: if `send_mail()` returns any relevant info, make sure to log it. Also try/except?

**Review by r-vieira — APPROVED** [review]
looks good. minor suggestions inline.

File: src/pkg/caendr/caendr/services/cloud/lifesciences.py (hunk 1 above)
Comment (vlagrassa) [reply src/pkg/caendr/caendr/services/cloud/lifesciences.py]: Fixed in [559920b](https://github.com/AndersenLab/CAENDR/pull/273/commits/559920bb87fb97040bb2b4624acb47f2d24b4e67)

File: src/pkg/caendr/caendr/services/cloud/lifesciences.py (hunk 2 above)
Comment (vlagrassa) [reply src/pkg/caendr/caendr/services/cloud/lifesciences.py]: Fixed in [559920b](https://github.com/AndersenLab/CAENDR/pull/273/commits/559920bb87fb97040bb2b4624acb47f2d24b4e67)

File: src/pkg/caendr/caendr/services/cloud/lifesciences.py (hunk 3 above)
Comment (vlagrassa) [reply src/pkg/caendr/caendr/services/cloud/lifesciences.py:195]: Fixed in [c1759eb](https://github.com/AndersenLab/CAENDR/pull/273/commits/c1759eb17cea4472307e2cd3817bb19fcad6c731)

Calling function does error handling / logging.

**Pull request merged**